WcaOnRails/app/models/registration.rb

```diff
@@ -86,6 +83,10 @@ def best_solve(event, type)
     person ? person.best_solve(event, type) : SolveTime.new(event.id, type, 0)
   end
 
+  def event_picker_events
+    registration_events.map(&:event_object).sort_by(&:rank)
+  end
```

Comment:
@jonatanklosko @larspetrus For organizers and delgates we deal with this by handling delagate_ids and organizer_ids in the model. I could do away with the accepts nested attributes and copy how it's been done for these fields. Let me know your thoughts.